Scalar.UnitTests/Common/GitVersionTests.cs

```diff
@@ -199,11 +244,57 @@ public void Allow_Invalid_Minor_Revision()
             version.Major.ShouldEqual(1);
             version.Minor.ShouldEqual(2);
             version.Build.ShouldEqual(3);
+            version.ReleaseCandidate.ShouldEqual(null);
             version.Platform.ShouldEqual("test");
             version.Revision.ShouldEqual(4);
             version.MinorRevision.ShouldEqual(0);
         }
 
+        [TestCase]
+        public void Allow_ReleaseCandidate()
+        {
+            GitVersion version;
+            GitVersion.TryParseVersion("1.2.3-rc4", out version).ShouldEqual(true);
+
+            version.Major.ShouldEqual(1);
+            version.Minor.ShouldEqual(2);
+            version.Build.ShouldEqual(3);
+            version.ReleaseCandidate.ShouldEqual(4);
+            version.Platform.ShouldBeNull();
+            version.Revision.ShouldEqual(0);
+            version.MinorRevision.ShouldEqual(0);
+        }
+
+        [TestCase]
+        public void Allow_ReleaseCandidate_Platform()
+        {
+            GitVersion version;
+            GitVersion.TryParseVersion("1.2.3-rc4.test", out version).ShouldEqual(true);
+
+            version.Major.ShouldEqual(1);
+            version.Minor.ShouldEqual(2);
+            version.Build.ShouldEqual(3);
+            version.ReleaseCandidate.ShouldEqual(4);
+            version.Platform.ShouldEqual("test");
+            version.Revision.ShouldEqual(0);
+            version.MinorRevision.ShouldEqual(0);
+        }
+
+        [TestCase]
+        public void Allow_LocalGitBuildVersion_ParseMajorMinorBuildOnly()
+        {
+            GitVersion version;
+            GitVersion.TryParseVersion("1.2.3.456.abcdefg.hijk", out version).ShouldEqual(true);
```

Comment:
nit: IIRC the local build string will be a 'g' followed by a series of hex digits..

Reply:
I'm not parsing this part if it fails to be an integer, so it doesn't really matter if it starts with a `g` or not IMO.
The key is that the `TryParseVersion` method returns `true`, and still parses the first 4 components.